test/test_auth.py

```diff
@@ -201,3 +202,149 @@ def test_scram_both(bouncer):
 
     # SCRAM password in userlist.txt
     bouncer.test(dbname="p62", user="scramuser1", password="foo")
+
+
+@pytest.mark.skipif("WINDOWS", reason="Windows does not have SIGHUP")
+def test_auth_dbname_usage(
+    bouncer,
+):
+    """
+    Check that the pgbouncer handles correctly the reserved pgbouncer
+    database usage as an authentication database
+    """
+
+    config = f"""
+        [databases]
+        pgbouncer_test = host={bouncer.pg.host} port={bouncer.pg.port} auth_dbname=pgbouncer
+        * = host={bouncer.host} port={bouncer.port} auth_dbname=pgbouncer
+        [pgbouncer]
+        auth_query = SELECT usename, passwd FROM pg_shadow where usename = $1
+        auth_user = pswcheck
+        stats_users = stats
+        listen_addr = {bouncer.host}
+        admin_users = pswcheck
+        auth_type = md5
+        auth_file = {bouncer.auth_path}
+        listen_port = {bouncer.port}
+        logfile = {bouncer.log_path}
+    """
+
+    # We expect that stats user does not exist in userlist.txt
+    with bouncer.log_contains(
+        'cannot use the reserved "pgbouncer" database as an auth_dbname', 3
+    ):
+        with bouncer.run_with_config(config):
+            #     Check the pgbouncer does not crash when we connect to pgbouncer admin db
+            with pytest.raises(psycopg.OperationalError, match="bouncer config error"):
+                bouncer.test(user="stats", password="stats", dbname="pgbouncer")
+
+            #     Check the pgbouncer does not crash when explicitly pgbouncer database
+            #     (admin DB) was set in auth_dbname in the databases definition section
+            with pytest.raises(psycopg.OperationalError, match="bouncer config error"):
+                bouncer.test(user="stats", password="stats", dbname="pgbouncer_test")
+
+            #     Check the pgbouncer does not crash when explicitly pgbouncer database
+            #     (admin DB) was set in auth_dbname in the databases definition section
+            with pytest.raises(psycopg.OperationalError, match="bouncer config error"):
+                bouncer.test(user="stats", password="stats", dbname="pgbouncer_test")
```

Comment:
It seems this currently tests the same as the the test above. I think you mean to use a different database name here, so we would trigger the wildcard case.

```suggestion
                bouncer.test(user="stats", password="stats", dbname="some_other_db")
```

Reply:
Agree! I provided the `p4` DB so that it now triggers the wildcard case